src/server/handler.go

```diff
@@ -180,7 +181,14 @@ func (fs FileSystem) Filecmd(request *sftp.Request) error {
 
 	switch request.Method {
 	case "Setstat":
-		var mode os.FileMode = 0644
+		var mode os.FileMode
+
+		if request.Attributes().FileMode().Perm() != 0000 {
+			mode = request.Attributes().FileMode().Perm()
+		} else {
+			mode = 0644
```

Comment:
Nitpick, but can you do `var mode os.FileMode = 0644;` and just have the if with no else here?

Reply:
No. This makes it so you can change the perms after it's been uploaded.

The other way made it so you couldn't update perms.